src/test/regress/sql/text_search.sql

```diff
@@ -0,0 +1,208 @@
+CREATE SCHEMA text_search;
+CREATE SCHEMA text_search2;
+SET search_path TO text_search;
+
+-- create a new configruation from scratch
+CREATE TEXT SEARCH CONFIGURATION my_text_search_config ( parser = default );
```

Comment:
Given that creation logic of Citus local tables are a little different than dist / ref table, it would be useful to create one citus local table and check if the shard also has the index:
```SQL

...
-- require coordinator in the metadata
SELECT citus_add_local_table_to_metadata('t1');


\d t1
              Table "text_search.t1"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 id     | integer |           |          | 
 name   | text    |           |          | 
Indexes:
    "t1_search_name" gin (to_tsvector('my_text_search_config'::regconfig, COALESCE(name, ''::character varying::text)))
\d t1_102009
           Table "text_search.t1_102009"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 id     | integer |           |          | 
 name   | text    |           |          | 
Indexes:
    "t1_search_name_102009" gin (to_tsvector('my_text_search_config'::regconfig, COALESCE(name, ''::character varying::text)))

```

We could consider `undistribute_table` as well as it also has some edge cases

Reply:
I wonder if this should be part of this PR.
This PR doesn't implement the creation of the indexes, that was already done correct.
Instead this PR purely adds the propagation of the object to remote machines.

Not saying this shouldn't be tested, but I guess this should be part of a more holistic test suite that tests all possible index creation syntax